sunbeam-python/sunbeam/plugins/interface/v1/tls.py

```diff
@@ -0,0 +1,251 @@
+        self.app = "keystone"
+        self.model = OPENSTACK_MODEL
+
+    def is_skip(self, status: Optional[Status] = None) -> Result:
+        """Determines if the step should be skipped or not.
+
+        :return: ResultType.SKIPPED if the Step should be skipped,
+                ResultType.COMPLETED or ResultType.FAILED otherwise
+        """
+        action_cmd = "list-ca-certs"
+        unit = run_sync(self.jhelper.get_leader_unit(self.app, self.model))
+
+        if not unit:
+            LOG.debug(f"Unable to get {self.app} leader")
+            return Result(ResultType.FAILED)
+
+        action_result = run_sync(self.jhelper.run_action(unit, self.model, action_cmd))
+        LOG.debug(f"Result from action {action_cmd}: {action_result}")
+
+        if action_result.get("return-code", 0) > 1:
+            LOG.debug(f"keystone action {action_cmd} returned error")
+            return Result(ResultType.FAILED)
+
+        action_result.pop("return-code")
+        ca_list = action_result
+        if self.name in ca_list:
+            return Result(ResultType.SKIPPED)
+
+        return Result(ResultType.COMPLETED)
+
+    def run(self, status: Optional[Status] = None) -> Result:
+        """Run keystone add-ca-certs action."""
+        action_cmd = "add-ca-certs"
+        unit = run_sync(self.jhelper.get_leader_unit(self.app, self.model))
+
+        if not unit:
+            LOG.debug(f"Unable to get {self.app} leader")
+            return Result(ResultType.FAILED)
+
+        action_params = {
+            "name": self.name,
+            "ca": self.ca_cert,
+            "chain": self.ca_chain,
+        }
+
+        LOG.debug(f"Running action {action_cmd} with params {action_params}")
+        action_result = run_sync(
+            self.jhelper.run_action(unit, self.model, action_cmd, action_params)
+        )
+        LOG.debug(f"Result from action {action_cmd}: {action_result}")
+
+        if action_result.get("return-code", 0) > 1:
+            LOG.debug(f"keystone action {action_cmd} returned error")
+            return Result(ResultType.FAILED)
+
+        return Result(ResultType.COMPLETED)
+
+
+class RemoveCACertsFromKeystoneStep(BaseStep):
+    """Remove CA certificates"""
+
+    def __init__(
+        self,
+        jhelper: JujuHelper,
+        name: str,
+    ):
+        super().__init__(
+            "Remove CA certs from keystone", "Removing CA certificates from keystone"
+        )
+        self.jhelper = jhelper
+        self.name = name.lower()
+        self.app = "keystone"
+        self.model = OPENSTACK_MODEL
+
+    def is_skip(self, status: Optional[Status] = None) -> Result:
+        """Determines if the step should be skipped or not.
+
+        :return: ResultType.SKIPPED if the Step should be skipped,
+                ResultType.COMPLETED or ResultType.FAILED otherwise
+        """
+        action_cmd = "list-ca-certs"
+        unit = run_sync(self.jhelper.get_leader_unit(self.app, self.model))
```

Comment:
get_leader_unit raises LeaderNotFoundException

Reply:
I do see couple of places that handle it as a None response instead of an exception...

Reply:
Fixed and pushed the changes.